File: app/src/main/java/net/sunniwell/app/linktaro/tools/PagerScroller.java
package net.sunniwell.app.linktaro.tools;

import android.content.Context;
import android.view.animation.Interpolator;
import android.widget.Scroller;

public class PagerScroller extends Scroller {
    private int mDuration = 1500;

    public PagerScroller(Context context) {
        super(context);
    }

    public PagerScroller(Context context, Interpolator interpolator) {
        super(context, interpolator);
    }

    public void startScroll(int startX, int startY, int dx, int dy, int duration) {
        super.startScroll(startX, startY, dx, dy, this.mDuration);
    }

    public void startScroll(int startX, int startY, int dx, int dy) {
        super.startScroll(startX, startY, dx, dy, this.mDuration);
    }

    public void setmDuration(int time) {
        this.mDuration = time;
    }

    public int getmDuration() {
        return this.mDuration;
    }
}

File: app/src/main/java/com/eliving/tv/customview/PagerScroller.kt
package com.eliving.tv.customview

import android.content.Context
import android.view.animation.Interpolator
import android.widget.Scroller

class PagerScroller : Scroller {
    private var mDuration = 1500

    constructor(context: Context?) : super(context) {}
    constructor(
        context: Context?,
        interpolator: Interpolator?
    ) : super(context, interpolator) {
    }

    override fun startScroll(
        startX: Int,
        startY: Int,
        dx: Int,
        dy: Int,
        duration: Int
    ) {
        super.startScroll(startX, startY, dx, dy, mDuration)
    }

    override fun startScroll(startX: Int, startY: Int, dx: Int, dy: Int) {
        super.startScroll(startX, startY, dx, dy, mDuration)
    }

    fun setmDuration(time: Int) {
        mDuration = time
    }

    fun getmDuration(): Int {
        return mDuration
    }
}